src/main/scala/org/scalameter/Executor.scala

```diff
@@ -30,21 +32,22 @@ object Executor {
   import Key._
 
   type Warmer = org.scalameter.Warmer
-  type Measurer = org.scalameter.Measurer
+  type Measurer[T] = org.scalameter.Measurer[T]
 
   val Warmer = org.scalameter.Warmer
   val Measurer = org.scalameter.Measurer
 
-  object None extends Executor {
-    def measurer: Measurer = ???
+  object None extends Executor[Nothing] {
+    def measurer: Measurer[Nothing] = ???
 
     def warmer: Warmer = ???
 
-    def runSetup[T](setup: Setup[T]): CurveData = ???
+    def runSetup[T](setup: Setup[T]): CurveData[Nothing] = ???
   }
 
-  trait Factory[E <: Executor] {
-    def apply(warmer: Warmer, aggregator: Aggregator, m: Measurer): E
+  trait Factory[E[_] <: Executor[_]] {
+    def apply[T: Numeric: Pickler](warmer: Warmer,
```

Comment:
About the `Numeric` - agreed, but `Pickler` is unfortunately needed. Both `LocalExecutor` and `SeparateJvmsExecutor` are creating `Measurement` so they require `Pickler` to be available in the scope.